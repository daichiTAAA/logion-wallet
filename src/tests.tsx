import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };
