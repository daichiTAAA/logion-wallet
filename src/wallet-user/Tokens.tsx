import React, { useState } from 'react';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

import { FullWidthPane } from '../common/Dashboard';
import Frame from '../common/Frame';
import Tabs from '../common/Tabs';

import { useCommonContext } from "../common/CommonContext";
import MyTokens from './MyTokens';
import Tokenization from "./Tokenization";
import PendingTokenizationRequests from './PendingTokenizationRequests';
import TokenizationRequestsHistory from './TokenizationRequestsHistory';

export default function Tokens() {
    const { colorTheme } = useCommonContext();
    const [ tabKey, setTabKey ] = useState<string>('pending');

    return (
        <FullWidthPane
            mainTitle="Tokens"
            titleIcon={{
                icon: {
                    id: 'tokens'
                },
                background: colorTheme.topMenuItems.iconGradient,
            }}
        >
            <Row>
                <Col>
                    <Frame
                        title="My tokens"
                    >
                        <MyTokens />
                        <Tokenization/>
                    </Frame>
                </Col>
            </Row>
            <Row>
                <Col>
                    <Frame
                        title="My requests"
                    >
                        <Tabs
                            activeKey={ tabKey }
                            onSelect={ key => setTabKey(key || 'pending') }
                            tabs={[
                                {
                                    key: "pending",
                                    title: "Pending",
                                    render: () => <PendingTokenizationRequests />
                                },
                                {
                                    key: "history",
                                    title: "History",
                                    render: () => <TokenizationRequestsHistory />
                                }
                            ]}
                        />
                    </Frame>
                </Col>
            </Row>
        </FullWidthPane>
    );
}
