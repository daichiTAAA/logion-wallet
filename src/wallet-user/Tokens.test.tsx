jest.mock('../common/CommonContext');

import React from "react";
import {shallowRender} from "../tests";
import Tokens from "./Tokens";

test("renders", () => {
    const tree = shallowRender(<Tokens/>)
    expect(tree).toMatchSnapshot();
});
