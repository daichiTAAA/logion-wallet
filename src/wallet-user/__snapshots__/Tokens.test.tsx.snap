// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<FullWidthPane
  mainTitle="Tokens"
  titleIcon={
    Object {
      "background": Object {
        "from": "#3b6cf4",
        "to": "#6050dc",
      },
      "icon": Object {
        "id": "tokens",
      },
    }
  }
>
  <Row
    noGutters={false}
  >
    <Col>
      <Frame
        title="My tokens"
      >
        <MyTokens />
        <Tokenization />
      </Frame>
    </Col>
  </Row>
  <Row
    noGutters={false}
  >
    <Col>
      <Frame
        title="My requests"
      >
        <Tabs
          activeKey="pending"
          onSelect={[Function]}
          tabs={
            Array [
              Object {
                "key": "pending",
                "render": [Function],
                "title": "Pending",
              },
              Object {
                "key": "history",
                "render": [Function],
                "title": "History",
              },
            ]
          }
        />
      </Frame>
    </Col>
  </Row>
</FullWidthPane>
`;
